modules/azure/storage.go

```diff
@@ -0,0 +1,246 @@
+)
+
+// StorageAccountExists indicates whether the storage account name exactly matches; otherwise false.
+func StorageAccountExists(t *testing.T, storageAccountName string, resourceGroupName string, subscriptionID string) bool {
+	result, err := StorageAccountExistsE(storageAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return result
+}
+
+// StorageBlobContainerExists returns true if the container name exactly matches; otherwise false
+func StorageBlobContainerExists(t *testing.T, containerName string, storageAccountName string, resourceGroupName string, subscriptionID string) bool {
+	result, err := StorageBlobContainerExistsE(containerName, storageAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return result
+}
+
+// GetStorageBlobContainerPublicAccess indicates whether a storage container has public access; otherwise false.
+func GetStorageBlobContainerPublicAccess(t *testing.T, containerName string, storageAccountName string, resourceGroupName string, subscriptionID string) bool {
+	result, err := GetStorageBlobContainerPublicAccessE(containerName, storageAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return result
+}
+
+// GetStorageAccountKind returns one of Storage, StorageV2, BlobStorage, FileStorage, or BlockBlobStorage.
+func GetStorageAccountKind(t *testing.T, storageAccountName string, resourceGroupName string, subscriptionID string) string {
+	result, err := GetStorageAccountKindE(storageAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return result
+}
+
+// GetStorageAccountSkuTier returns the storage account sku tier as Standard or Premium.
+func GetStorageAccountSkuTier(t *testing.T, storageAccountName string, resourceGroupName string, subscriptionID string) string {
+	result, err := GetStorageAccountSkuTierE(storageAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return result
+}
+
+// GetStorageDNSString builds and returns the storage account dns string if the storage account exists.
+func GetStorageDNSString(t *testing.T, storageAccountName string, resourceGroupName string, subscriptionID string) string {
+	result, err := GetStorageDNSStringE(storageAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return result
+}
+
+// StorageAccountExistsE indicates whether the storage account name exactly matches; otherwise false.
+// This function would fail the test if there is an error.
+func StorageAccountExistsE(storageAccountName, resourceGroupName, subscriptionID string) (bool, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return false, err2
+	}
+	storageAccount, err3 := GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID)
+	if err3 != nil {
```

Comment:
Does this need to follow pattern to check for resourcenotfound @davesee for comment.

	if err != nil {
		if ResourceNotFoundErrorExists(err) {
			return false, nil
		}
		return false, err
	}
	return true, nil

Reply:
done